data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/DataPrepperExtensionPoints.java

```diff
@@ -12,13 +12,17 @@
 
 import javax.inject.Inject;
 import javax.inject.Named;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
 
 @Named
 public class DataPrepperExtensionPoints implements ExtensionPoints {
     private static final ExtensionProvider.Context EMPTY_CONTEXT = new EmptyContext();
     private final GenericApplicationContext sharedApplicationContext;
     private final GenericApplicationContext coreApplicationContext;
+    private Map<Class, Optional<Object>> providers;
```

Comment:
I think a `Set<Class<? extends ExtensionProvider>>` should be sufficient. You don't need the value and you can avoid the call to `extensionProvider.provideInstance(EMPTY_CONTEXT)`.